libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioClientWrapperOptions.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright(c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Bot.Builder.Adapters.Twilio
+{
+    /// <summary>
+    /// Defines options for a <see cref="TwilioClientWrapper"/>.
+    /// </summary>
+    public class TwilioClientWrapperOptions
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TwilioClientWrapperOptions"/> class.
+        /// </summary>
+        /// <param name="twilioNumber">The twilio phone number.</param>
+        /// <param name="twilioAccountSid">The account id.</param>
+        /// <param name="twilioAuthToken">The authentication token.</param>
+        /// <param name="twilioValidationUrl">The validation URL for incoming requests.</param>
+        public TwilioClientWrapperOptions(string twilioNumber, string twilioAccountSid, string twilioAuthToken, Uri twilioValidationUrl = null)
+        {
+            if (string.IsNullOrWhiteSpace(twilioNumber))
+            {
+                throw new ArgumentException($"The {nameof(twilioNumber)} property for {nameof(TwilioAdapterOptions)} can't be empty.", nameof(twilioNumber));
+            }
+
+            if (string.IsNullOrWhiteSpace(twilioAccountSid))
+            {
+                throw new ArgumentException($"The {nameof(twilioNumber)} property for {nameof(TwilioAdapterOptions)} can't be empty.", nameof(twilioAccountSid));
+            }
+
+            if (string.IsNullOrWhiteSpace(twilioAuthToken))
+            {
+                throw new ArgumentException($"The {nameof(twilioAuthToken)} property for {nameof(TwilioAdapterOptions)} can't be empty.", nameof(twilioAuthToken));
+            }
+
+            TwilioNumber = twilioNumber;
+            TwilioAccountSid = twilioAccountSid;
+            TwilioAuthToken = twilioAuthToken;
+            TwilioValidationUrl = twilioValidationUrl;
+        }
+
+        /// <summary>
+        /// Gets or sets the phone number associated with this Twilio app.
+        /// </summary>
+        /// <value>
+        /// The phone number, in the format 1XXXYYYZZZZ.
```

Comment:
This is a USA Formatted phone number. OTher countries have different formats. Is this US Specific?

Reply:
Removed US specific comment.